src/bp/core/services/bot-service.ts

```diff
@@ -451,4 +465,75 @@ export class BotService {
     BotService._mountedBots.forEach((isMounted, bot) => isMounted && bots.push(bot))
     return bots
   }
+
+  public async listRevisions(botId: string): Promise<string[]> {
+    const globalGhost = this.ghostService.global()
+
+    let stageID = ''
+    if (await this.workspaceService.hasPipeline()) {
+      const botConfig = await this.configProvider.getBotConfig(botId)
+      stageID = botConfig.pipeline_status.current_stage.id
+    }
+
+    const revisions = await globalGhost.directoryListing(REVISIONS_DIR)
+
+    return revisions.filter(rev => rev.includes(botId) && rev.includes(stageID)).sort((revA, revB) => {
+      const dateA = revA.split('::')[1].replace('.tgz', '')
+      const dateB = revB.split('::')[1].replace('.tgz', '')
+      const diff = moment(dateA, 'YY-MM-DD-ha').diff(moment(dateB, 'YY-MM-DD-ha'))
```

Comment:
I'd use a constant for the moment() format, since you use it in different places and they must match to list / create revisions. What do you think of using a different format, YYYYMMDD-hhmmss ? 'ha' isn't specific enough, like '12pm' . We could skip the 'ss' part tho, but would prevent someone from deleting a previous revision in the same minute

Reply:
I wanted this to be readable without performing any date manipulation in the front end + I thought 1 revision / hour is enough